packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -931,6 +941,49 @@ class _CupertinoDatePickerDateState extends State<CupertinoDatePicker> {
 // layout will be broken.
 
 
+/// Controls a [CupertinoTimerPicker] widget.
+///
+/// Each controller can only be used with a single Cupertino timer picker.
+///
+/// Used with [CupertinoTimerPicker].
+///
+/// See also:
+///
+///  * [CupertinoTimerPicker] which creates a Cupertino themed time picker which
+///    uses this controller.
+///  * [CupertinoDatePicker] which creates a Cupertino themed date picker.
+///
+class CupertinoTimerPickerController extends ChangeNotifier {
+  /// Creates a CupertinoTimerPickerController.
+  ///
+  /// The [resetAnimationDuration] and [desiredTimerDuration] arguments must not be null.
+  CupertinoTimerPickerController({
+    Duration desiredTimerDuration = Duration.zero,
+    this.resetAnimationDuration = const Duration(milliseconds: 300),
+  }) : assert(resetAnimationDuration != null),
+       assert(desiredTimerDuration != null) {
+    _desiredTimerDuration = desiredTimerDuration;
+  }
+
+  /// The desired duration of the countdown timer.
+  ///
+  /// Setting this duration to any value will animate the picker back to the
+  /// given duration.
+  ///
+  /// Defaults to Duration.zero.
+  Duration get desiredTimerDuration => _desiredTimerDuration;
+  Duration _desiredTimerDuration;
+  set desiredTimerDuration(Duration duration) {
```

Comment:
timer is redundant since the class is timer picker controller. This whole thing can just be called `duration`.